### DISCUSSION TITLE
NuxtLink not hydrated from <component> tag

### DISCUSSION THREAD

**Discussion by silencerspirit** [nuxt/nuxt#17067] [Questions] [answered]
Can't resolve NuxtLink from `<component is="NuxtLink" />`

Here is reproduction https://stackblitz.com/edit/nuxt-starter-qqvjt7?file=app.vue

**Comment by danielroe** [comment, marked as answer]
As with any auto-imported component, you will need to run `const NuxtLink = resolveComponent('NuxtLink')` in your script setup if you want to use it programmatically.

https://v3.nuxtjs.org/guide/directory-structure/components/#dynamic-components
